### ISSUE TITLE
Better to have offload method to create createMetricReporter()

### ISSUE THREAD

**Issue by ruks** [wso2/apim-analytics-publisher#19]
- Currently createMetricReporter get class name as argument and use default if null pass
- Better to have another createMetricReporter() without class name as argument

ex:
```
public MetricReporter createMetricReporter(String fullyQualifiedClassName, Map<String, String> properties) throws MetricCreationException {
   this.createMetricReporter("org.wso2.am.analytics.publisher.reporter.ChoreoAnalyticsMetricReporter", properties);
}
```

**Comment by tishan89** [comment]
+1 I also found it weird to have a null. Also gonna introduce builders and a log metric reporter

**Comment by tishan89** [comment]
Fixed with 4db3c111bc1840bf930dd22f14cdbb1fdca3fc1a
